### PR TITLE
refactor(app): add hook for overflow menu outside click

### LINKED ISSUE #10086: 6.0 Feedback:  Protocols > Protocol Detail overflow menu persists after no selection
### Overview

When I open the overflow menu on a protocol detail it does not disappear when I click off of it.  The only way it collapses without selecting an item in the menu is to click the overflow menu kebab again.

### Current Behavior

https://user-images.githubusercontent.com/502770/165359299-cf1e30aa-8570-4042-9a47-a9c9ff5c0fb5.mp4

### Expected Behavior

Clicking off of the the menu or clicking the kebab would close the menu.

### Steps To Reproduce

See the video.

### Operating system

Mac

### Robot setup or anything else?

_No response_

### LINKED ISSUE #10209: 6.0 Feedback: Open menu overflow appears above of a slideout on T1 Protocols
### Overview

Menu overflow appears above of the slideout on T1 Protocols when you click 'import'

### Current Behavior

[Loom video](https://www.loom.com/share/b17be4e6e91d4dc5b6062aacdf8e21bb)

### Expected Behavior

The menu should be below the slideout, or automatically closes when you perform another action (like open another menu, or click the import button)

### Steps To Reproduce

n/a

### Operating system

Mac

### Robot setup or anything else?

n/a

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR abstracts the logic into a hook in order to hide the overflow menu when an outside node is clicked.

closes #10086, closes #10209

# Changelog

- Added `useMenuHandleClickOutside`

# Review requests

- Check that when an overflow menu is clicked and then clicking outside of the menu closes it.

# Risk assessment

low

### CODE REVIEW

File: app/src/atoms/MenuList/hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    e.preventDefault()
+    setShowOverflowMenu(!showOverflowMenu)
+  }
+
+  const handleClickOutside: React.MouseEventHandler<HTMLDivElement> = e => {
+    e.preventDefault()
+    setShowOverflowMenu(false)
+  }
+
+  const MenuOverlayPortal = (): JSX.Element => (
```
Comment (sakibh) [inline app/src/atoms/MenuList/hooks.tsx]: Any suggestions for naming this component properly?

**Review by koji — APPROVED** [review]
Looks good to me.
It's working as expected.

`OverflowMenuOverlayPortal` sounds tricky to me, so MenuOverlayPortal would be okay but let's ask this to the team on Slack. Maybe someone has a good idea.

File: app/src/organisms/Devices/RobotOverflowMenu.tsx [hunk 2]
```diff
@@ -144,26 +140,19 @@ export function RobotOverflowMenu(props: RobotOverflowMenuProps): JSX.Element {
...
-          />
-        ) : null}
-        {showConnectionTroubleshootingModal ? (
-          <ConnectionTroubleshootingModal
-            onClose={() => setShowConnectionTroubleshootingModal(false)}
-          />
-        ) : null}
-      </Portal>
+      <MenuOverlayPortal />
+      {robot.status === CONNECTABLE ? (
```
Comment (b-cooper) [inline app/src/organisms/Devices/RobotOverflowMenu.tsx]: it looks like these two modals were rendered in the portal before, but now they aren't, was this change intentional?

File: app/src/organisms/ProtocolDetails/ProtocolLabwareDetails.tsx [hunk 3]
```diff
@@ -192,20 +186,13 @@ export const LabwareDetailOverflowMenu = (
...
-        ) : null}
-        {showLabwareDetailSlideout ? (
-          <LabwareDetails
-            labware={labware}
-            onClose={() => setShowLabwareDetailSlideout(false)}
-          />
-        ) : null}
-      </Portal>
+      <MenuOverlayPortal />
+      {showLabwareDetailSlideout ? (
```
Comment (b-cooper) [inline app/src/organisms/ProtocolDetails/ProtocolLabwareDetails.tsx]: this slideout was also being rendered into the Portal, but is not now, was that change intended?

File: app/src/organisms/Devices/RobotOverflowMenu.tsx (hunk 2 above)
Comment (sakibh) [reply app/src/organisms/Devices/RobotOverflowMenu.tsx]: I added an optional `children` prop so this should now render inside the Portal again.

File: app/src/organisms/ProtocolDetails/ProtocolLabwareDetails.tsx (hunk 3 above)
Comment (sakibh) [reply app/src/organisms/ProtocolDetails/ProtocolLabwareDetails.tsx]: I added an optional `children` prop so this should now render inside the Portal again.

File: app/src/organisms/Devices/RobotOverflowMenu.tsx (hunk 2 above)
Comment (b-cooper) [reply app/src/organisms/Devices/RobotOverflowMenu.tsx]: Won't that lead to the OverflowMenu's overlay also getting rendered, with its click handler and all, when either of the slideout or connection modal are rendered? 

Also nit: they're not "menu's", even though they're being rendered into something called "MenuOverlayPortal". I'd probably just return the same thing but named `MenuOverlay` from the hook and not concern the user of this hook with the underlying implementation of that overlay which involves a `Portal`.  Then I'd add back in the `Portal` that was wrapping both the slideout and the connect troubleshooting modal.  What do you think?

**Review by b-cooper — APPROVED** [review]
This looks good, just noticed that the `MenuIOverlay` component returned from the hook is wrapped in an unnecessary React Fragment. Feel free to address this in a follow up if it's easier/
